llvm/test/Transforms/InstCombine/or-xor.ll

```diff
@@ -261,9 +261,12 @@ define i32 @test10_commuted(i32 %A, i32 %B) {
 
 define i32 @test10_extrause(i32 %A, i32 %B, ptr %dst) {
 ; CHECK-LABEL: @test10_extrause(
-; CHECK-NEXT:    [[NOT:%.*]] = xor i32 [[A:%.*]], -1
+; CHECK-NEXT:    [[XOR1:%.*]] = xor i32 [[B:%.*]], [[A:%.*]]
+; CHECK-NEXT:    [[NOT:%.*]] = xor i32 [[A]], -1
 ; CHECK-NEXT:    store i32 [[NOT]], ptr [[DST:%.*]], align 4
-; CHECK-NEXT:    ret i32 -1
+; CHECK-NEXT:    [[XOR2:%.*]] = xor i32 [[B]], [[NOT]]
+; CHECK-NEXT:    [[OR:%.*]] = or i32 [[XOR1]], [[XOR2]]
+; CHECK-NEXT:    ret i32 [[OR]]
```

Comment:
regression + below

Reply:
Fixed in https://github.com/llvm/llvm-project/commit/534701d5f93369e822f3afc9670e3a42b08dfc6f.